Embedded/MaxMix/Config.h

```diff
@@ -39,16 +46,18 @@ static const uint8_t  STATE_GAME_COUNT = 3;
 static const uint8_t  STATE_DISPLAY_AWAKE = 0;
 static const uint8_t  STATE_DISPLAY_SLEEP = 1;
 
-// --- Rotary Encoder Acceleration
-// The acceleration is computed from a simple line equation (y = mx + b)
-// at 150ms, there should be no acceleration (1 increment)
-static const float ROTARY_ACCELERATION_SLOW_TIME = 150;
-static const float ROTARY_ACCELERATION_SLOW_INC = 1;
-// at 25ms, we want to have maximum acceleration (16 increments)
-static const float ROTARY_ACCELERATION_FAST_TIME = 25;
-static const float ROTARY_ACCELERATION_FAST_INC = 16;
-static const float ROTARY_ACCELERATION_M_SLOPE = (ROTARY_ACCELERATION_FAST_INC - ROTARY_ACCELERATION_SLOW_INC) / (ROTARY_ACCELERATION_FAST_TIME - ROTARY_ACCELERATION_SLOW_TIME);
-static const float ROTARY_ACCELERATION_B_OFFSET = ROTARY_ACCELERATION_SLOW_INC - ROTARY_ACCELERATION_M_SLOPE * ROTARY_ACCELERATION_SLOW_TIME;
+// --- Rotary Encoder
+#define WITHOUT_BUTTON 1
+// Increasing this will reduce the acceleration effect
+// Values between 100 - 600 work well
+static const uint16_t ROTARY_ACCELERATION_DIVISOR = 400;
+
+// Half Stepping
+#ifdef HALF_STEP
+static const uint8_t ROTARY_PULSES_PER_STEP = 2;
+#else
+static const uint8_t ROTARY_PULSES_PER_STEP = 4;
+#endif
```

Comment:
I wonder if this shouldn't be a config value in the app that is sent down. Kinda sucks as a user needing to custom build firmware just because you ended up with a 2 pulse encoder instead of 4 pulse encoder.

Reply:
Yea good point. Should have this as an option too. I wanted to get some feedback if this feels better than the current linear acceleration.

Reply:
https://github.com/t3knomanzer/maxmix-software/blob/2027ab18bc1189c2eb5d76319d49c6821dddf2b8/Desktop/Application/MaxMix/Services/Communication/MessageSettings.cs#L43-L65

How do I add a 2-byte value to the array (uint16_t)? The acceleration value will exceed 1 byte.

Reply:
It'd be great for the steps to be a setting but if it's too much trouble, I am currently compiling 2 versions of the software for every release and the user just needs to pick the one they want from the firmware installer, so no trouble at all for the end user really.

Reply:
I think it's easier to reconfigure the stepping during runtime for ClickEncoder since it's just a divisor, while Rotary uses a table. We could load both tables in fw and the just follow different tables for each stepping.